sstables/generation_type.hh

```diff
@@ -31,6 +31,17 @@ public:
 
     constexpr bool operator==(const generation_type& other) const noexcept { return _value == other._value; }
     constexpr std::strong_ordering operator<=>(const generation_type& other) const noexcept { return _value <=> other._value; }
+
+    generation_type& operator++() noexcept {
+        ++_value;
+        return *this;
+    }
```

Comment:
I'd be happier with a method new_generation somewhere (not in generation_type). Note it depends on the table's settings, not just on what the latest generation looks like.

Reply:
yeah, i thought about this also. it's like a move-only input iterator -- we can only read from it for a new value. the `value_type` of this iterator is `generation_type`. but they are not of the same type.

Reply:
Yes, generation_type should be restricted to storing info about a single generation. All the coordination should be performed in replica::table, possibly with some guidance from sstable_manager (as to what generation type should be used). sstable_manager also guides replica::table as to what is the highest sstable version supported.